crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs

```diff
@@ -508,33 +545,82 @@ impl KitsuneP2pHandler for Space {
         let fut = self.i_s.update_agent_info();
         let i_s = self.i_s.clone();
         let evt_sender = self.evt_sender.clone();
-        let bootstrap_service = self.config.bootstrap_service.clone();
-        if let Some(bootstrap_service) = bootstrap_service {
-            tokio::task::spawn(async move {
-                const START_DELAY: std::time::Duration = std::time::Duration::from_secs(1);
-                const MAX_DELAY: std::time::Duration = std::time::Duration::from_secs(60 * 60);
-                let mut delay_len = START_DELAY;
-
-                loop {
-                    tokio::time::delay_for(delay_len).await;
-                    if delay_len <= MAX_DELAY {
-                        delay_len *= 2;
-                    }
-
-                    if let Err(e) = super::discover::add_5_or_less_non_local_agents(
-                        space.clone(),
-                        agent.clone(),
-                        i_s.clone(),
-                        evt_sender.clone(),
-                        bootstrap_service.clone(),
-                    )
-                    .await
-                    {
-                        tracing::error!(msg = "Failed to get peers from bootstrap", ?e);
+        match self.config.network_type {
+            NetworkType::QuicMdns => {
+                // Listen to MDNS service that has that space as service type
+                let space_b64 = base64::encode_config(&space[..], base64::URL_SAFE_NO_PAD);
+                //println!("(MDNS) - Agent {:?} ({}) joined space {:?} ({} ; {})", agent, agent.get_bytes().len(), space, space.get_bytes().len(), dna_str.len());
+                tokio::task::spawn(async move {
+                    let stream = mdns_listen(space_b64);
+                    tokio::pin!(stream);
+                    while let Some(maybe_response) = stream.next().await {
+                        match maybe_response {
+                            Ok(response) => {
+                                tracing::debug!(msg = "Peer found via MDNS", ?response);
```

Comment:
How frequently does this happen? Debug level is good so long as it isn't flooding the other debug logs : )